--- libevent-2.1.4-alpha-old/sample/include.am	2014-09-01 15:54:36.997722734 +0200
+++ libevent-2.1.4-alpha/sample/include.am	2014-09-01 15:54:42.884590763 +0200
@@ -33,8 +33,6 @@
 
 $(SAMPLES) : libevent.la
 
-sample_event_read_fifo_SOURCES = sample/event-read-fifo.c
-sample_event_read_fifo_LDADD = $(LIBEVENT_GC_SECTIONS) libevent.la
 sample_time_test_SOURCES = sample/time-test.c
 sample_time_test_LDADD = $(LIBEVENT_GC_SECTIONS) libevent.la
 sample_signal_test_SOURCES = sample/signal-test.c
--- libevent-2.1.4-alpha-old/Makefile.in	2014-09-01 15:54:36.977712985 +0200
+++ libevent-2.1.4-alpha/Makefile.in	2014-09-01 16:53:19.689825752 +0200
@@ -296,7 +296,6 @@
 @OPENSSL_TRUE@am__EXEEXT_2 = sample/le-proxy$(EXEEXT) \
 @OPENSSL_TRUE@	sample/https-client$(EXEEXT)
 am__EXEEXT_3 = sample/dns-example$(EXEEXT) \
-	sample/event-read-fifo$(EXEEXT) sample/hello-world$(EXEEXT) \
 	sample/http-server$(EXEEXT) sample/signal-test$(EXEEXT) \
 	sample/time-test$(EXEEXT) $(am__EXEEXT_2)
 am__EXEEXT_4 = test/bench$(EXEEXT) test/bench_cascade$(EXEEXT) \
@@ -312,10 +311,6 @@
 am_sample_dns_example_OBJECTS = sample/dns-example.$(OBJEXT)
 sample_dns_example_OBJECTS = $(am_sample_dns_example_OBJECTS)
 sample_dns_example_DEPENDENCIES = $(am__DEPENDENCIES_1) libevent.la
-am_sample_event_read_fifo_OBJECTS = sample/event-read-fifo.$(OBJEXT)
-sample_event_read_fifo_OBJECTS = $(am_sample_event_read_fifo_OBJECTS)
-sample_event_read_fifo_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-	libevent.la
 am_sample_hello_world_OBJECTS = sample/hello-world.$(OBJEXT)
 sample_hello_world_OBJECTS = $(am_sample_hello_world_OBJECTS)
 sample_hello_world_DEPENDENCIES = $(am__DEPENDENCIES_1) libevent.la
@@ -467,7 +462,6 @@
 SOURCES = $(libevent_la_SOURCES) $(libevent_core_la_SOURCES) \
 	$(libevent_extra_la_SOURCES) $(libevent_openssl_la_SOURCES) \
 	$(libevent_pthreads_la_SOURCES) $(sample_dns_example_SOURCES) \
-	$(sample_event_read_fifo_SOURCES) \
 	$(sample_hello_world_SOURCES) $(sample_http_server_SOURCES) \
 	$(sample_https_client_SOURCES) $(sample_le_proxy_SOURCES) \
 	$(sample_signal_test_SOURCES) $(sample_time_test_SOURCES) \
@@ -484,7 +478,6 @@
 	$(am__libevent_openssl_la_SOURCES_DIST) \
 	$(am__libevent_pthreads_la_SOURCES_DIST) \
 	$(sample_dns_example_SOURCES) \
-	$(sample_event_read_fifo_SOURCES) \
 	$(sample_hello_world_SOURCES) $(sample_http_server_SOURCES) \
 	$(am__sample_https_client_SOURCES_DIST) \
 	$(am__sample_le_proxy_SOURCES_DIST) \
@@ -1046,7 +1039,7 @@
 @INSTALL_LIBEVENT_TRUE@include_event2_HEADERS = $(EVENT2_EXPORT)
 @INSTALL_LIBEVENT_TRUE@nodist_include_event2_HEADERS = include/event2/event-config.h
 @INSTALL_LIBEVENT_FALSE@nodist_noinst_HEADERS = include/event2/event-config.h
-SAMPLES = sample/dns-example sample/event-read-fifo sample/hello-world \
+SAMPLES = sample/dns-example sample/hello-world \
 	sample/http-server sample/signal-test sample/time-test \
 	$(am__append_6)
 @OPENSSL_TRUE@sample_le_proxy_SOURCES = sample/le-proxy.c
@@ -1059,8 +1052,6 @@
 
 @OPENSSL_TRUE@sample_https_client_LDADD = libevent.la libevent_openssl.la ${OPENSSL_LIBS} ${OPENSSL_LIBADD}
 @OPENSSL_TRUE@sample_https_client_INCLUDES = $(OPENSSL_INCS)
-sample_event_read_fifo_SOURCES = sample/event-read-fifo.c
-sample_event_read_fifo_LDADD = $(LIBEVENT_GC_SECTIONS) libevent.la
 sample_time_test_SOURCES = sample/time-test.c
 sample_time_test_LDADD = $(LIBEVENT_GC_SECTIONS) libevent.la
 sample_signal_test_SOURCES = sample/signal-test.c
@@ -1354,12 +1347,6 @@
 sample/dns-example$(EXEEXT): $(sample_dns_example_OBJECTS) $(sample_dns_example_DEPENDENCIES) $(EXTRA_sample_dns_example_DEPENDENCIES) sample/$(am__dirstamp)
 	@rm -f sample/dns-example$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(sample_dns_example_OBJECTS) $(sample_dns_example_LDADD) $(LIBS)
-sample/event-read-fifo.$(OBJEXT): sample/$(am__dirstamp) \
-	sample/$(DEPDIR)/$(am__dirstamp)
-
-sample/event-read-fifo$(EXEEXT): $(sample_event_read_fifo_OBJECTS) $(sample_event_read_fifo_DEPENDENCIES) $(EXTRA_sample_event_read_fifo_DEPENDENCIES) sample/$(am__dirstamp)
-	@rm -f sample/event-read-fifo$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(sample_event_read_fifo_OBJECTS) $(sample_event_read_fifo_LDADD) $(LIBS)
 sample/hello-world.$(OBJEXT): sample/$(am__dirstamp) \
 	sample/$(DEPDIR)/$(am__dirstamp)
 
@@ -1605,7 +1592,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/strlcpy.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/win32select.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@sample/$(DEPDIR)/dns-example.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@sample/$(DEPDIR)/event-read-fifo.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@sample/$(DEPDIR)/hello-world.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@sample/$(DEPDIR)/hostcheck.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@sample/$(DEPDIR)/http-server.Po@am__quote@
--- libevent-2.1.4-alpha-old/http.c	2014-09-01 15:54:36.977712985 +0200
+++ libevent-2.1.4-alpha/http.c	2014-09-01 16:59:04.149495307 +0200
@@ -137,8 +137,10 @@
 				return (0);
 		} else {
 			struct hostent *hp;
-			hp = gethostbyaddr((char *)&sin->sin_addr,
-			    sizeof(struct in_addr), AF_INET);
+			fprintf(stderr,"gethostbyaddr not supported by NaCl. Will exit.\n");
+			exit(1);
+			//hp = gethostbyaddr((char *)&sin->sin_addr,
+			//    sizeof(struct in_addr), AF_INET);
 			if (hp == NULL)
 				return (-2);
 
--- libevent-2.1.4-alpha-old/test/test-ratelim.c	2014-09-01 15:54:37.009728583 +0200
+++ libevent-2.1.4-alpha/test/test-ratelim.c	2014-09-01 17:05:42.054001442 +0200
@@ -113,11 +113,7 @@
 	struct client_state *cs = ctx;
 	struct evbuffer *output = bufferevent_get_output(bev);
 	char buf[1024];
-#ifdef _WIN32
 	int r = rand() % 256;
-#else
-	int r = random() % 256;
-#endif
 	memset(buf, r, sizeof(buf));
 	while (evbuffer_get_length(output) < 8192) {
 		evbuffer_add(output, buf, sizeof(buf));
@@ -587,11 +583,7 @@
 	{
 		struct timeval tv;
 		evutil_gettimeofday(&tv, NULL);
-#ifdef _WIN32
 		srand(tv.tv_usec);
-#else
-		srandom(tv.tv_usec);
-#endif
 	}
 
 #ifndef EVENT__DISABLE_THREAD_SUPPORT
--- libevent-2.1.4-alpha-old/test/test-time.c	2014-09-01 15:54:37.009728583 +0200
+++ libevent-2.1.4-alpha/test/test-time.c	2014-09-01 17:06:49.146688394 +0200
@@ -51,11 +51,7 @@
 static int
 rand_int(int n)
 {
-#ifdef _WIN32
 	return (int)(rand() % n);
-#else
-	return (int)(random() % n);
-#endif
 }
 
 static void
--- libevent-2.1.4-alpha-old/test/regress.c	2014-09-01 15:54:37.009728583 +0200
+++ libevent-2.1.4-alpha/test/regress.c	2014-09-01 20:38:49.955416578 +0200
@@ -946,6 +946,7 @@
 static void
 test_simplesignal(void)
 {
+#if 0
 	struct event ev;
 	struct itimerval itv;
 
@@ -968,11 +969,13 @@
 		test_ok = 0;
 
 	cleanup_test();
+#endif
 }
 
 static void
 test_multiplesignal(void)
 {
+#if 0
 	struct event ev_one, ev_two;
 	struct itimerval itv;
 
@@ -999,6 +1002,7 @@
 		test_ok = 0;
 
 	cleanup_test();
+#endif
 }
 
 static void
diff --git a/signal.c b/signal.c
index 3f46295..19adb7e 100644
--- a/signal.c
+++ b/signal.c
@@ -126,6 +126,15 @@ evsig_set_base_(struct event_base *base)
 	EVSIGBASE_UNLOCK();
 }
 
+#ifdef __native_client__
+#  ifndef SA_RESTART
+#    define SA_RESTART 0
+#  endif
+#  ifndef SA_NODEFER
+#    define SA_NODEFER 0
+#  endif
+#endif
+
 /* Callback for when the signal handler write a byte to our signaling socket */
 static void
 evsig_cb(evutil_socket_t fd, short what, void *arg)
diff --git a/configure b/configure
index 8a5e46d..dc226ed 100755
--- a/configure
+++ b/configure
@@ -13033,7 +13033,7 @@ case "$enable_openssl" in
      yes) ;;
      *)
 	save_LIBS="$LIBS"
-	LIBS=""
+	#LIBS=""
 	OPENSSL_LIBS=""
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for library containing SSL_new" >&5
 $as_echo_n "checking for library containing SSL_new... " >&6; }
diff --git a/configure b/configure
index dc226ed..efc4240 100755
--- a/configure
+++ b/configure
@@ -15336,7 +15336,7 @@ else
 
 fi
 ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
diff --git a/configure b/configure
index efc4240..21f4558 100755
--- a/configure
+++ b/configure
@@ -2603,7 +2603,7 @@ fi
 ## -------------------- ##
 
 ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
@@ -3293,7 +3293,7 @@ fi
 
 
 ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
@@ -4076,7 +4076,7 @@ if test "x$ac_cv_prog_cc_c89" != xno; then :
 fi
 
 ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
@@ -4211,7 +4211,7 @@ fi
 
 
 ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
@@ -4342,7 +4342,7 @@ See \`config.log' for more details" "$LINENO" 5; }
 fi
 
 ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
@@ -7386,7 +7386,7 @@ if ${lt_cv_cc_needs_belf+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
@@ -7410,7 +7410,7 @@ fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
      ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
@@ -8633,7 +8633,7 @@ esac
 
 lt_save_CC="$CC"
 ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
@@ -12214,7 +12214,7 @@ $as_echo "$enable_static" >&6; }
 
 fi
 ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
@@ -14998,7 +14998,7 @@ if test x$bwin32 != xtrue && test "$enable_thread_support" != "no"; then
 
 
 ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnudiff --git a/bufferevent_sock.c b/bufferevent_sock.c
diff --git a/http.c b/http.c
index c5e143a..84f6132 100644
--- a/http.c
+++ b/http.c
@@ -43,6 +43,7 @@
 #endif
 
 #ifndef _WIN32
+#include <sys/cdefs.h>
 #include <sys/resource.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
